### PR TITLE
[WIP] Move auctions API requests into GraphQL

### PR DESCRIPTION
**Pull request by ashfurrow**
Fix for the performance issues described in https://github.com/artsy/auctions/issues/353

Requires https://github.com/artsy/metaphysics/pull/608 and https://github.com/artsy/metaphysics/pull/609 to be merged first.

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">No CHANGELOG changes made</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_travis">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Artsy/Models/API_Models/Auctions/Bid.m [hunk 1]
```diff
@@ -24,12 +23,12 @@ - (BOOL)isEqual:(id)object
 
 - (BOOL)isEqualToBid:(Bid *)otherBid
 {
-    return [self.bidID isEqual:otherBid.bidID];
+    return [self.cents isEqual:otherBid.cents];
```
Comment (orta) [inline Artsy/Models/API_Models/Auctions/Bid.m]: Are you sure on this? Couldn't two people make a bid at the same price?

File: Artsy/Networking/ARRouter.m [hunk 2]
```diff
@@ -1073,11 +1073,127 @@ + (NSURLRequest *)liveSaleStateRequest:(NSString *)saleID host:(NSString *)host
...
+          cm\
+          in\
+        }\
+        display: title\
+        sold: is_sold\
+      }\
+    }\
+  }\
+}\
+    }", saleID];
```
Comment (orta) [inline Artsy/Networking/ARRouter.m]: that's one hell of a query

File: Artsy/Models/API_Models/Auctions/Bid.m (hunk 1 above)
Comment (ashfurrow) [inline Artsy/Models/API_Models/Auctions/Bid.m]: I'll double check, but I believe we only compare bids from different users on the same lot. Good catch though.

File: Artsy/Networking/ARRouter.m (hunk 2 above)
Comment (alloy) [inline Artsy/Networking/ARRouter.m]: Heh. Might want to move to a `auctionData.graphql` file –even if only for the syntax highlighting– and make it a real GraphQL query which means you don’t need to interpolate yourself.

From the JS example at the end of [this page](http://graphql.org/graphql-js/passing-arguments/):

```js
xhr.send(JSON.stringify({
  query: query,
  variables: { dice: dice, sides: sides },
}));
```

**Comment by ashfurrow** [comment]
Closing in favour of #2278.

**Pull request closed without merging**